mobile/src/bundles/talent/screens/bsa-badges.tsx

```diff
@@ -0,0 +1,94 @@
+import { type NavigationProp } from '@react-navigation/native';
+import React from 'react';
+
+import { View } from '~/bundles/common/components/components';
+import {
+    type TalentOnboardingScreenName,
+    TalentOnboardingScreenNumber,
+    TalentOnboardingStepState,
+} from '~/bundles/common/enums/enums';
+import {
+    useAppDispatch,
+    useAppRoute,
+    useAppSelector,
+    useCallback,
+    useNavigation,
+} from '~/bundles/common/hooks/hooks';
+import { globalStyles } from '~/bundles/common/styles/global-styles';
+import {
+    type TalentOnboardingNavigationParameterList,
+    type ValueOf,
+} from '~/bundles/common/types/types';
+import {
+    BsaBadgesForm,
+    NewAccountHeader,
+} from '~/bundles/talent/components/components';
+import { getNextStepTitle } from '~/bundles/talent/helpers/helpers';
+import { actions as talentActions } from '~/bundles/talent/store';
+import { type BsaBadgesStepDto } from '~/bundles/talent/types/types';
+
+const BsaBadges: React.FC = () => {
+    const { name } = useAppRoute();
+    const dispatch = useAppDispatch();
+    const { onboardingData } = useAppSelector(({ talents }) => talents);
+
+    const { currentUserData } = useAppSelector(({ auth }) => auth);
+    const userId = currentUserData?.id ?? '';
+
+    const badgesStepData: BsaBadgesStepDto | null = onboardingData?.badges
+        ? {
+              badges: onboardingData.badges,
+          }
+        : null;
+
+    const stepTitle = name as ValueOf<typeof TalentOnboardingScreenName>;
+    const stepNumber = TalentOnboardingScreenNumber[stepTitle];
+
+    const { navigate } =
+        useNavigation<
+            NavigationProp<TalentOnboardingNavigationParameterList>
+        >();
+
+    const handleSubmit = useCallback(
+        async (payload: BsaBadgesStepDto): Promise<void> => {
+            const updatedBadgesPayload = {
+                ...payload,
+                userId,
+            };
+
+            const result = await dispatch(
+                talentActions.updateOnboardingData(updatedBadgesPayload),
+            );
+
+            if (result.payload) {
+                const setStepResult = dispatch(
+                    talentActions.setCompletedStep(stepTitle),
+                );
```

Comment:
will we get this completedStep value from the backend in the future?

Reply:
yes, we discussed that there will be a field in user-details with the name of the last completed step. 
For now i made it only in state, maybe for navigation.

Reply:
Not number of step, but title